File: entrada/archivo1.c
#include <stdio.h>
#include <stdbool.h>
#include "archivo3.h"
#include "archivo2.h"

bool esPrimo(int num) {
    if (num <= 1) return false;
    for (int i = 2; i * i <= num; i++) {
        if (num % i == 0) return false;
    }
    return true;
}

int main() {
    int numero;
    printf("Introduce un número para verificar si es primo: ");
    scanf("%d", &numero);

    if (esPrimo(numero)) {
        printf("%d es un número primo.\n", numero);
    } else {
        printf("%d no es un número primo.\n", numero);
    }

    return 0;
}

File: salida/archivo1.rs
fn esPrimo (i32  num) -> bool {{
		if (num <= 1){
			1
		}
		for i in 2..= num{
			if (num % i == 0){
				0
			}
		}
		true
	}
	
	fn () main-> i32 {
		let numero:i32 ;
		
		
		print!("Introduce un número para verificar si es primo: ");
		let mut input = String::new();
		io::stdin()
		.read_line(&mut input)
		.expect("Failed to read line");
		if (esPrimo(numero)){
			print!("%d es un número primo.\n{}",numero);
		}
		else{
				print!("%d no es un número primo.\n{}",numero);}
		
		0
	}
	
	